custom_components/ai_thermostat/climate.py

```diff
@@ -671,16 +688,22 @@ async def _async_control_heating(self):
 					_LOGGER.warning("ai_thermostat: call for heat decision: could not evaluate sensor/weather entity data, force heat on")
 					self.call_for_heat = True
 				
-				_LOGGER.debug("ai_thermostat %s",self.call_for_heat)
 				# window open detection and weather detection force turn TRV off
-				if (self.window_open or not self.call_for_heat) and not self.closed_window_triggered:
-					self.heating_active_pre_window_opened = False
-					if self._hvac_mode == HVAC_MODE_HEAT:
-						self.heating_active_pre_window_opened = True
+				if self.window_open and not self.closed_window_triggered:
```

Comment:
We should initialize `self.window_open` with `None` and check here that it is indeed already filled with something sensible and we're not parsing the init value of the variable.